daemons/fenced/pacemaker-fenced.c

```diff
@@ -619,16 +619,15 @@ static void
 update_stonith_watchdog_timeout_ms(xmlNode *cib)
 {
     xmlNode *stonith_enabled_xml = NULL;
-    const char *stonith_enabled_s = NULL;
+    bool stonith_enabled = false;
+    int rc = pcmk_rc_ok;
     long timeout_ms = 0;
 
     stonith_enabled_xml = get_xpath_object("//nvpair[@name='stonith-enabled']",
                                            cib, LOG_NEVER);
-    if (stonith_enabled_xml) {
-        stonith_enabled_s = crm_element_value(stonith_enabled_xml, XML_NVPAIR_ATTR_VALUE);
-    }
+    rc = pcmk__xe_get_bool_attr(stonith_enabled_xml, XML_NVPAIR_ATTR_VALUE, &stonith_enabled);
 
-    if (stonith_enabled_s == NULL || crm_is_true(stonith_enabled_s)) {
+    if (rc == EINVAL || rc == ENODATA || (rc == pcmk_rc_ok && stonith_enabled)) {
```

Comment:
I think this boils down to "if (rc != pcmk_rc_ok || stonith_enabled)" since we want to default to true, which means unknown_format should hit as well. That would be a change from the previous behavior but I think it's a fix.

Reply:
At the least, this change doesn't do anything to cts-scheduler ouput, which lets me think it's probably reasonable.